.github/workflows/test_objectivec.yml

```diff
@@ -68,17 +68,23 @@ jobs:
     strategy:
       fail-fast: false   # Don't cancel all jobs if one fails.
       matrix:
-        PLATFORM: ["ios", "macos", "tvos", "watchos"]
+        PLATFORM: ["ios", "macos", "tvos", "watchos", "visionos"]
         CONFIGURATION: ["Debug", "Release"]
-    name: CocoaPods ${{ matrix.PLATFORM}} ${{ matrix.CONFIGURATION}}
-    runs-on: macos-12
-    env:
-      DEVELOPER_DIR: /Applications/Xcode_14.1.app/Contents/Developer
+        include:
+          - OS: macos-12
+          - OS: macos-14
+            PLATFORM: "visionos"
+            XCODE: Xcode_15.2
+    name: CocoaPods ${{ matrix.PLATFORM }} ${{ matrix.CONFIGURATION }}
+    runs-on: ${{ matrix.OS }}
     steps:
       - name: Checkout pending changes
         uses: protocolbuffers/protobuf-ci/checkout@v2
         with:
           ref: ${{ inputs.safe-checkout }}
+      - name: Xcode version
+        if: matrix.OS == 'macos-14'
```

Comment:
I think it would make sense to keep this unconditional and just pin a version in macos-12

Reply:
If I'm not misunderstanding your opinion:
This job not pin version in macos-12 before, so it dependency on the [default version](https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md#xcode) which the runner providered.

Sry about my poor English.

Reply:
I just wanna to do not change the behavior it is used to be, by just add visionOS pod-lib-lint check.